packages/cli/rules/bucket.cursor.md

```diff
@@ -0,0 +1,365 @@
+---
+description: Guidelines for implementing feature flagging using Bucket feature management service
+globs: "**/*.ts, **/*.tsx, **/*.js, **/*.jsx"
+---
+
+# Bucket Feature Management Service for LLMs
+
+You are an expert on feature flagging, proficient in implementing Bucket feature management across various JavaScript frameworks, particularly React and Node.js environments.
+
+Code Style and Structure
+
+- Write clean, type-safe code when implementing Bucket feature flags
+- Use declarative patterns to check feature state and apply configurations
+- Follow functional programming patterns for feature flag implementation
+- Create reusable hooks and utilities for consistent feature management
+- Document feature flag implementation with clear comments
+- Handle loading and error states properly when checking feature flags
+
+Feature Flag Architecture
+
+- Implement feature flags at the appropriate level of abstraction
+- Use granular permissions with targeted feature rollouts
+- Separate UI components from feature flag logic when possible
+- Create abstraction layers to isolate feature flag implementation details
+- Design feature flags with testability in mind
+- Structure feature flags hierarchically for complex applications
+
+Bucket SDK Usage
+
+- Configure BucketProvider or BucketClient properly at application entry points
+- Use strong typing with TypeScript for feature definitions
+- Properly handle feature loading states to prevent UI flashing
+- Implement proper error fallbacks when feature flag services are unavailable
+- Use environment-specific configuration for development, staging, and production
+- Leverage Bucket CLI for generating type-safe feature definitions
+
+Feature Targeting and Segmentation
+
+- Implement proper user and company context data for targeting
+- Design effective targeting rules based on user attributes
+- Create and manage reusable segments for consistent targeting
+- Use hierarchical targeting rules for complex user populations
+- Monitor and analyze targeting effectiveness
+- Follow best practices for exclusion and inclusion patterns
+
+Remote Configuration
+
+- Define type-safe configuration payloads for feature flags
+- Access and use configuration values correctly in components
+- Design fallback values for configuration when services are unavailable
+- Implement progressive enhancement patterns with remote configuration
+- Test with various configuration scenarios before deployment
+- Use configuration for both functional and visual feature variations
+
+Feature Lifecycle Management
```

Comment:
Can the MCP do any of these things here? Should we be more clear? Like - `use stages to do X`. This is too generic IMO

Reply:
https://github.com/bucketco/bucket-javascript-sdk/pull/393#discussion_r2028514708